### PR TITLE
#152 re-send publish request on timeout

### PR DESCRIPTION
**Pull request by georgschoelly**
Maybe part of a fix for #152

### CODE REVIEW

**Comment by oroulet** [comment]
have you tested it? should be possible if you set timeout very low in client and another server

**Comment by georgschoelly** [comment]
Not yet, but I will do so soon.

File: opcua/client/ua_client.py [hunk 1]
```diff
@@ -400,23 +400,32 @@ def publish(self, acks=None):
...
+            self._uasocket.check_answer(data, "while waiting for publish response")
+        except BadTimeout: # Spec Part 4, 7.28
+            self.publish()
+
         try:
             response = ua.PublishResponse.from_binary(data)
             self.logger.debug(response)
         except Exception:
             self.logger.exception("Error parsing notificatipn from server")
             self.publish([]) #send publish request ot server so he does stop sending notifications
```
Comment (georgschoelly) [inline opcua/client/ua_client.py]: Does anyone know in which cases this happens? It seems to me that catching `BadTimeout` has we now do above is the only case re-sending a publish request is appropriate.

Comment (oroulet) [inline opcua/client/ua_client.py]: not that I know..

**Comment by georgschoelly** [comment]
I just tested it and works as intended.

**Comment by oroulet** [comment]
OK, our server does not support it, but it does not care about timeout on requests either....

**Pull request merged**